test/test_datasets.py

```diff
@@ -33,6 +33,28 @@ def test_fashionmnist(self):
         self.assertTrue(isinstance(target, int))
         shutil.rmtree(tmp_dir)
 
+    def test_imagenet(self):
+        tmp_dir = tempfile.mkdtemp()
+        archive_dict = torchvision.datasets.imagenet.ARCHIVE_DICT
+        archive_dict['train']['url'] = ('https://github.com/pmeier/vision/blob/imagenet_test/'
```

Comment:
Can we instead of downloading this file patch (using `mock.patch`) `download_url` so that it returns the file present in `test/assets`?

Reply:
I need your advice here, since I have never worked with `mock` before. 

Suppose we patch `download_url` to simply copy the fake data to `tmp_dir`. How would we handle the download of the `devkit`? I think we cannot upload it together with the fake data due to licensing issues. Thus, I see two options:

1. Patch  `download_url` in a way that it retains its original functionality if called with an `url` that ends in `ILSVRC2012_devkit_t12.tar.gz`
2. Split `ImageNet.download()` in two methods (e. g. `download_devkit` and `download_images`) and only patch `download_url` after `download_devkit` was called. Unfortunately, this means we can't simply test `ImageNet(..., download=True)`, which maybe defeats the whole purpose of this test.

Do you have a better idea?